include/mailio/export.hpp

```diff
@@ -0,0 +1,26 @@
+/*
+
+mime.hpp
+--------
+
+Copyright (C) 2016, Tomislav Karastojkovic (http://www.alepho.com).
+
+Distributed under the FreeBSD license, see the accompanying file LICENSE or
+copy at http://www.freebsd.org/copyright/freebsd-license.html.
+
+*/
+
+#ifndef MAILIO_EXPORT_H
+#define MAILIO_EXPORT_H
+
+#if defined(MAILIO_BUILDING_DLL) || defined(MAILIO_USING_DLL)
+# if defined(MAILIO_BUILDING_DLL)
+#  define MAILIO_EXPORT __declspec(dllexport)
+# else
+#  define MAILIO_EXPORT __declspec(dllimport)
+# endif
+#else
+# define MAILIO_EXPORT
+#endif
+
+#endif
```

Comment:
I believe the DLL export should be for Windows only, something like
```
#pragma once

#ifdef _WIN32
#  if defined(MAILIO_BUILDING_DLL) || defined(MAILIO_USING_DLL)
#    if defined(MAILIO_BUILDING_DLL)
#      define MAILIO_EXPORT __declspec(dllexport)
#    else
#      define MAILIO_EXPORT __declspec(dllimport)
#    endif
#  endif
#else
#  define MAILIO_EXPORT
#endif
```
Otherwise, I cannot compile it on Linux (although did not test this on Windows yet).

Reply:
autotools can't generate a VS project file, but it can generate a Makefile that mingw will use